core/src/components/toast/animations/md.enter.ts

```diff
@@ -1,20 +1,59 @@
 import { createAnimation } from '@utils/animation/animation';
+import { win } from '@utils/browser';
 import { getElementRoot } from '@utils/helpers';
 
 import type { Animation } from '../../../interface';
+import type { ToastPresentOptions } from '../toast-interface';
 
 /**
  * MD Toast Enter Animation
  */
-export const mdEnterAnimation = (baseEl: HTMLElement, position: string): Animation => {
+export const mdEnterAnimation = (baseEl: HTMLElement, opts: ToastPresentOptions): Animation => {
   const baseAnimation = createAnimation();
   const wrapperAnimation = createAnimation();
+  const { position, positionAnchor } = opts;
 
   const root = getElementRoot(baseEl);
   const wrapperEl = root.querySelector('.toast-wrapper') as HTMLElement;
 
-  const bottom = `calc(8px + var(--ion-safe-area-bottom, 0px))`;
-  const top = `calc(8px + var(--ion-safe-area-top, 0px))`;
+  /**
+   * Start with a predefined offset from the edge the toast will be
+   * positioned relative to. By default, this will be the top edge
+   * of the screen for position="top", or the bottom edge for
+   * position="bottom".
+   */
+  let offset = 8;
+  let top, bottom;
+
+  /**
+   * If positionAnchor is defined, add in the distance from the target
+   * screen edge to the target anchor edge. For position="top", the
+   * bottom anchor edge is targeted. For position="bottom", the top
+   * anchor edge is targeted.
+   */
+  if (positionAnchor && win) {
+    const box = positionAnchor.getBoundingClientRect();
+    if (position === 'top') {
+      offset += box.bottom;
+    } else if (position === 'bottom') {
+      /**
+       * Just box.top is the distance from the top edge of the screen
+       * to the top edge of the anchor. We want to calculate from the
+       * bottom edge of the screen instead.
+       */
+      offset += win.innerHeight - box.top;
+    }
+
+    /**
+     * We don't include safe area here because that should already be
```

Comment:
Optional: We could move this to a utility function so both iOS/MD animations can use it. However, since it's only used twice this is optional.

Reply:
I decided against this because the two are subtly different -- they have different starting offsets and, more importantly, the `bottom` positions are generally calculated using addition for `md` but subtraction for `ios`. This is because `md` uses `style.bottom`, meaning the position is calculated from the bottom edge of the screen, but `ios` uses `translateY` which calculates from the top edge instead.

I could certainly have the util function check the `mode` and act accordingly, but the context behind the difference would sit outside the util, making it harder to follow the code. There's probably some refactoring that could be done here to align the two modes, but that feels outside the scope of this PR.

Reply:
Although, if I need to avoid implicit keyframes in the leave animation, in theory it would make more sense to do this calculation out in `toast.tsx` so I can pass it to both the enter and leave animations and save on calculations 🤔 Hmm. I'll play around with it some.

Reply:
I've gone with the above idea. Let me know if you think it'd be worth it to make a tech debt ticket to sync up how the toast's position is calculated between modes, so the new `getAnimationPosition` util can be cleaned up a little. I'm not sure if it's worth the effort since the position-related CSS would need touching too 😅

Reply:
I like your approach here. The additional benefit is if devs want to create a custom toast animation they can still take advantage of the anchor/position information since we pass that in.